src/Sylius/Bundle/RbacBundle/Security/RbacVoter.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\RbacBundle\Security;
+
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Sylius\Component\Rbac\Model\IdentityInterface;
+use Sylius\Component\Rbac\Authorization\PermissionMapInterface;
+use Sylius\Component\Rbac\Provider\PermissionProviderInterface;
+use Sylius\Component\Rbac\Resolver\RolesResolverInterface;
+use Sylius\Component\Rbac\Exception\PermissionNotFoundException;
+use Doctrine\Common\Cache\Cache;
+
+/**
+ * @author Christian Daguerre <christian@daguer.re>
+ */
+class RbacVoter implements RbacVoterInterface
+{
+    const DEFAULT_TTL = 60;
+
+    /**
+     * @var PermissionMapInterface
+     */
+    protected $permissionMap;
+
+    /**
+     * @var PermissionProviderInterface
+     */
+    protected $permissionProvider;
+
+    /**
+     * @var RolesResolverInterface
+     */
+    protected $rolesResolver;
+
+    /**
+     * @var Cache
+     */
+    protected $permissionsCache;
+
+    /**
+     * @var int
+     */
+    protected $ttl;
+
+    public function __construct(
+        PermissionMapInterface $permissionMap,
+        PermissionProviderInterface $permissionProvider,
+        RolesResolverInterface $rolesResolver,
+        Cache $permissionsCache,
+        $ttl = self::DEFAULT_TTL
+    ) {
+        $this->permissionMap      = $permissionMap;
```

Comment:
It's my personal feel, but if we use another, longer variable here, we will have to modify all the assignments.